ompi/mca/coll/inter/coll_inter_allgatherv.c

```diff
@@ -94,16 +97,31 @@ mca_coll_inter_allgatherv_inter(const void *sbuf, int scount,
             ptmp = ptmp_free - gap;
 	}
     }
+    OMPI_COUNT_ARRAY_INIT(&count_arg, count);
+    OMPI_DISP_ARRAY_INIT(&displace_arg, displace);
     err = comm->c_local_comm->c_coll->coll_gatherv(sbuf, scount, sdtype,
-						  ptmp, count, displace,
-						  sdtype,0, comm->c_local_comm,
-                                                  comm->c_local_comm->c_coll->coll_gatherv_module);
+                                                   ptmp, &count_arg, &displace_arg,
+                                                   sdtype,0, comm->c_local_comm,
+                                                   comm->c_local_comm->c_coll->coll_gatherv_module);
     if (OMPI_SUCCESS != err) {
         goto exit;
     }
 
-    ompi_datatype_create_indexed(size,rcounts,disps,rdtype,&ndtype);
+    /* TODO:BIGCOUNT: Remove tehese temporaries once ompi_datatype is updated for bigcount */
+    int *tmp_rcounts = malloc(sizeof(int) * size);
+    int *tmp_disps = malloc(sizeof(int) * size);
+    if (NULL == tmp_rcounts || NULL == tmp_disps) {
+        err = OMPI_ERR_OUT_OF_RESOURCE;
+        goto exit;
+    }
+    for (i = 0; i < size; ++i) {
+        tmp_rcounts[i] = (int) ompi_count_array_get(rcounts, i);
+        tmp_disps[i] = (int) ompi_disp_array_get(disps, i);
+    }
+    ompi_datatype_create_indexed(size,tmp_rcounts,tmp_disps,rdtype,&ndtype);
```

Comment:
Is there a PR for the datatype interface? I wonder whether we can reuse the array structure there?

Reply:
There's a draft PR: #12351. It seems like these structs could be used there as well.